app/javascript/lib/validation/shared-validation.js

```diff
@@ -186,3 +188,23 @@ export function validateGroupColumnCounts(headers, line, isLastLine, dataObj) {
   }
   return issues
 }
+
+/**
+ * Timeout the CSFV if taking longer than 10 seconds
+ *
+ */
+export function timeOutCSFV(startTime, chunker) {
+  const maxTime = 10000 // in milliseconds this equates to 10 seconds
+  const maxRealTime = startTime + maxTime
+  const currentTime = new Date().getTime()
+  const issues = []
+
+  if (currentTime > maxRealTime) {
+    // quit early by setting the file reader to the end of the file so it can't read anymore
+    chunker.setToFileEnd()
+    issues.push(['warn', 'timeout',
+      'Due to the size of the file, validation will occur after upload. Please be aware '+
+        'the absence of errors/warnings here is NOT an indication of the validity of the file.'])
```

Comment:
Briefer phrasing, with a worthwhile additional note, and a lower-volume "not".

```suggestion
      'Due this file\'s size, it will be fully validated after upload, and results will be emailed to you.  ' + 
      'The absence of issues here does not mean the file is valid.  ' +
```